### DISCUSSION TITLE
asyncio stream writer or  reader udp client

### DISCUSSION THREAD

**Discussion by skylin008** [micropython/micropython#13207] [STM32 / Pyboard]
Is there a asyncio stream writer or reader udp client example to be used.Thanks!

**Comment by jimmo** [comment]
@skylin008 Unfortunately UDP in asyncio is not yet supported.

**Comment by peterhinch** [comment]
You can access UDP at socket level using nonblocking sockets, but I don't think anyone has done much work on this. See [this simple example](https://github.com/peterhinch/micropython-mqtt/blob/88c0b971fec268fcb397d991de4daadfd169dbac/mqtt_as/mqtt_as.py#L324).

**Comment by karfas** [comment]
You might also find https://github.com/karfas/upy-simple-app/blob/master/lib%2Fasy_udp_client.py a useful starting point.

Be aware this is work in progress, so some features might not work as expected.

**Reply by skylin008** [reply]
Thanks @karfas  @peterhinch
